### PR TITLE
feat(events): Batch event insertion

### LINKED ISSUE #310: Add support for Insights API
### Feature Description
The [go-insights](https://github.com/newrelic/go-insights) client + CLI supports the [Insights API](https://docs.newrelic.com/docs/insights/insights-api). It could be potentially beneficial to include this functionality in the client so that other applications can utilize it, such as **newrelic-cli**.

This ticket is loosely related to a [request that was submitted](https://github.com/newrelic/go-insights/issues/16) in go-insights.

### PR DESCRIPTION
**Pull request by jthurman42**
Add ability to start a background queue of events, and batch post them to New Relic.

Porting from newrelic/go-insights with some updates / modifications.

Closes #310

### CODE REVIEW

File: pkg/events/example_basic_test.go [hunk 1]
```diff
@@ -32,3 +33,36 @@ func Example_basic() {
 		log.Fatal("error posting custom event:", err)
 	}
 }
+
+func Example_batch() {
```
Comment (ctrombley) [inline pkg/events/example_basic_test.go:37]: Can you put this in its own file (`example_batch_test.go`?) so the godoc will render it as a separate use case?

Comment (jthurman42) [reply pkg/events/example_basic_test.go:37]: yeah good call.

File: pkg/events/events_batch.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+		return errors.New("queueing not enabled for this client")
+	}
+
+	jsonData, err := e.marshalEvent(event)
+	if err != nil {
+		return err
+	}
+
+	select {
+	case e.eventQueue <- *jsonData:
```
[the remark this answers is not in the document]
Comment (jthurman42) [reply pkg/events/events_batch.go:114]: marshalEvent always returns an error with a nil response, but I can add a check above for that

File: pkg/events/events_batch.go [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+// This is also used by watchdog when the timer expires.
+func (e *Events) Flush() error {
+	if e.flushQueue == nil {
+		return errors.New("queueing not enabled for this client")
+	}
+
+	e.logger.Debug("flushing events")
+
+	for x := 0; x < e.batchWorkers; x++ {
+		e.flushQueue[x] <- true
```
[the remark this answers is not in the document]
Comment (jthurman42) [reply pkg/events/events_batch.go:132]: shouldn't happen, but I'll change this to a `range` to simplify the code

**Pull request merged**